### PR TITLE
CLN: trim unreachable groupby paths

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -266,7 +266,11 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
             func = maybe_mangle_lambdas(func)
             ret = self._aggregate_multiple_funcs(func)
             if relabeling:
-                ret.columns = columns
+                # error: Incompatible types in assignment (expression has type
+                # "Optional[List[str]]", variable has type "Index")
+                ret.columns = columns  # type: ignore[assignment]
+            return ret
+
```
Comment (jreback) [inline pandas/core/groupby/generic.py:273]: you could outdent (and remove the else) here to make it more obvious

Comment (jbrockmendel) [reply pandas/core/groupby/generic.py:273]: considered that, but this mirrors the layout of the DataFrameGroupBy method and im holding out hope of sharing more of that

File: pandas/core/groupby/generic.py [hunk 2]
```diff
@@ -1000,6 +998,11 @@ def aggregate(self, func=None, *args, engine=None, engine_kwargs=None, **kwargs)
         result = op.agg()
         if not is_dict_like(func) and result is not None:
             return result
+        elif relabeling and result is not None:
+            # this should be the only (non-raising) case with relabeling
```
Comment (jbrockmendel) [inline pandas/core/groupby/generic.py:1002]: @rhshadrach i wrote this comment a few years ago and now need help figuring it out.  in particular, can re rule out `relabeling and result is None`?  Or is that a shortcoming of the test suite?

Comment (rhshadrach) [reply pandas/core/groupby/generic.py:1002]: Yes - it is not possible that `relabeling` is True and `result` is None. If `relabeling` is True, the `func` returned from `reconstruct_func` must be the result from `normalize_keyword_aggregation` which is a dictionary. In the case of a dict-like, the result is always computed in `Apply.agg` via `Apply.agg_dict_like`.

Comment (jbrockmendel) [reply pandas/core/groupby/generic.py:1002]: so this condition can be replaced with `elif relabling:`?

Comment (rhshadrach) [reply pandas/core/groupby/generic.py:1002]: Yes. Also, as an aside, I think it was a misstep to try to reuse the code in `core.apply` for `groupby.agg`. I don't have any concrete ideas for cleaning this up yet, but it's on my radar.

**Pull request merged**